industrial_ci/src/tests/source_tests.sh

```diff
@@ -78,6 +78,8 @@ file) # When UPSTREAM_WORKSPACE is file, the dependended packages that need to b
     elif [ -e $TARGET_REPO_PATH/$ROSINSTALL_FILENAME ]; then
         # install (maybe unreleased version) dependencies from source
         $ROSWS merge file://$TARGET_REPO_PATH/$ROSINSTALL_FILENAME
+    else
+        error "UPSTREAM_WORKSPACE file '$TARGET_REPO_PATH/$ROSINSTALL_FILENAME[.$ROS_DISTRO]' does not exist"
```

Comment:
> What's the purpose for enclosing the file extension as [.$ROS_DISTRO]?

At this point we can't tell which file is missing, the generic or the distro-specific one.